operators/pkg/tenant-controller/sandbox.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tenant_controller groups the functionalities related to the Tenant controller.
+package tenant_controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clctx "github.com/netgroup-polito/CrownLabs/operators/pkg/context"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/forge"
+	"github.com/netgroup-polito/CrownLabs/operators/pkg/utils"
+)
+
+const (
+	// StringSeparator -> the separator used to concatenate string.
+	StringSeparator = "-"
+)
+
+// EnforceSandboxResources ensures the presence/absence of a sandbox.
+func (r *TenantReconciler) EnforceSandboxResources(ctx context.Context) error {
+	tenant := clctx.TenantFrom(ctx)
+
+	if tenant.Spec.CreateSandbox {
+		err := r.enforceSandboxResourcesPresence(ctx)
+		if err != nil {
+			klog.Errorf("Unable to update namespace of tenant %s -> %s", tenant.Name, err)
+			tenant.Status.SandboxNamespace.Created = false
+			tenant.Status.SandboxNamespace.Name = ""
+			tnOpinternalErrors.WithLabelValues("tenant", "sandbox-resources").Inc()
+			return err
+		}
```

Comment:
You are not consistent, as you are managing here the creation errors but not the deletion ones. I would suggest to move the error management logic to the calling function.